src/sentry/api/endpoints/debug_files.py

```diff
@@ -177,23 +179,31 @@ def get(self, request: Request, project) -> Response:
         else:
             q = Q()
 
-        file_format_q = Q()
-        for file_format in file_formats:
-            known_file_format = DIF_MIMETYPES.get(file_format)
-            if known_file_format:
-                file_format_q |= Q(file__headers__icontains=known_file_format)
+        if file_formats:
+            file_format_q = Q()
+            for file_format in file_formats:
+                known_file_format = DIF_MIMETYPES.get(file_format)
+                if known_file_format:
+                    file_format_q |= Q(file__headers__icontains=known_file_format)
```

Comment:
This could get expensive as you're scanning un-indexed fields across what could be a large set of data for busy projects.

Reply:
You are right. Symbolicator removed the filter it previously had, massively reducing the number of requests it does through caching.
The Sentry UI still has one case where it provides a filter, the other one I cleaned up a while ago: https://github.com/getsentry/sentry/pull/45704/files

So this filter has existed for quite some time, and still has to for reasons. I don’t know of any specific case where this has actually caused problems previously.